src/boosting/dart.hpp

```diff
@@ -106,23 +105,30 @@ class DART: public GBDT {
         train_score_updater_->AddScore(models_[curr_tree].get(), curr_class);
       }
     }
-    shrinkage_rate_ = 1.0 / (1.0 + drop_index_.size());
+    shrinkage_rate_ = gbdt_config_->learning_rate / (gbdt_config_->learning_rate + static_cast<double>(drop_index_.size()));
   }
   /*!
   * \brief normalize dropped trees
+  * NOTE: num_drop_tree(k), learning_rate(lr), shrinkage_rate_ = lr / (k + lr)
+  *       step 1: shrink tree to -1 -> drop tree
+  *       step 2: shrink tree to k * lr / (k + lr) - 1 from -1
+  *               -> normalize for valid data
+  *       step 3: shrink tree to k * lr / (k + lr) from k * lr / (k + lr) - 1
+  *               -> normalize for train data
+  *       end with tree weight = k * lr / (k + lr)
```

Comment:
it's same as before. 
before:
weights of dropped trees: k / (1 + k)
weights of new trees:  1 / (1 + k)
now: 
change 1 to lr